File: Backend/src/main/java/com/sistema/examenes/sistema_examenes_backend/servicios/ComprobanteService.java
package com.sistema.examenes.sistema_examenes_backend.servicios;

import com.sistema.examenes.sistema_examenes_backend.Enums.EstadoComprobante;
import com.sistema.examenes.sistema_examenes_backend.entidades.Comprobante;

import java.util.List;

public interface ComprobanteService {

    public Comprobante crearComprobante(Comprobante comprobante);

    public List<Comprobante> obtenerComprobantes();

    //public List<Comprobante> obtenerComprobantesPorUsuario();

    public Comprobante obtenerComprobantePorId(Long id);

    public Comprobante actualizarComprobante(Comprobante comprobante);

    public Comprobante eliminarComprobante(Long id);

    public List<Comprobante> obtenerComprobantesPorCliente(Long clienteId);

    public List<Comprobante> obtenerComprobantesPorEstado(EstadoComprobante estadoComprobante);

    public Comprobante crearComprobantePorCita(Long citaId, Comprobante comprobante);






}

File: Frontend/src/app/services/comprobante.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ComprobanteService {

  constructor(private http: HttpClient) { }

  // Obtener todos los comprobantes
  obtenerComprobantes(): Observable<any[]> {
    return this.http.get<any[]>(`${baserUrl}/comprobantes`);
  }

  // Crear un nuevo comprobante
  crearComprobante(comprobante: any): Observable<any> {
    return this.http.post<any>(`${baserUrl}/comprobantes/crear`, comprobante);
  }

  // Obtener comprobantes por cliente
  obtenerComprobantesPorCliente(clienteId: number): Observable<any[]> {
    return this.http.get<any[]>(`${baserUrl}/comprobantes/cliente/${clienteId}`);
  }

  // Obtener comprobantes por estado
  obtenerComprobantesPorEstado(estado: string): Observable<any[]> {
    return this.http.get<any[]>(`${baserUrl}/comprobantes/estado/${estado}`);
  }

  // Crear comprobante por cita
  crearComprobantePorCita(citaId: number, comprobante: any): Observable<any> {
    return this.http.post<any>(`${baserUrl}/comprobantes/cita/${citaId}`, comprobante);
  }
}
